rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java

```diff
@@ -783,10 +796,25 @@ public TransactionReceiptDTO eth_getTransactionReceipt(String transactionHash) {
 
     @Override
     public BlockResultDTO eth_getUncleByBlockHashAndIndex(String blockHash, String uncleIdx) {
+        if(!HexUtils.hasHexPrefix(blockHash)) {
+            throw invalidParamError("invalid argument 0: cannot unmarshal hex string without 0x prefix into hash");
+        }
+
         BlockResultDTO s = null;
 
         try {
-            Block block = blockchain.getBlockByHash(stringHexToByteArray(blockHash));
+            byte[] bhash = stringEvenHexToByteArray(blockHash);
```

Comment:
same comment as in the `eth_getTransactionByHash` method. 
Additionally,  the validation of a transaction hash would be the same for almost any method. So I think we should extract the code to a validator class and create related unit tests instead of repeat it.

Reply:
Makes sense, I'll refactor this PR and implement the validator class approach Max uses.